flocker/apiclient/_client.py

```diff
@@ -0,0 +1,90 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Client for the Flocker REST API.
+"""
+
+from uuid import UUID
+
+from zope.interface import Interface
+
+from pyrsistent import PClass, field, pmap_field
+
+
+class Dataset(PClass):
```

Comment:
Ah no. Because ``flocker.control.Dataset`` has no notion of `primary`. It just belongs to a `Node`.